examples/src/main/scala/sttp/client3/examples/SimpleClientGetAndPost.scala

```diff
@@ -0,0 +1,32 @@
+package sttp.client3.examples
+
+import sttp.client3.{Request, Response, SttpClient, UriContext, asStringAlways, basicRequest}
+
+import java.util.UUID
+
+object SimpleClientGetAndPost extends App {
+  val client = SttpClient()
+
+  try {
+    val response: Response[Either[String, String]] = client.send(basicRequest.get(uri"https://httpbin.org/get"))
+
+    response.body match {
+      case Left(body)  => println(s"Non-2xx response to GET with code ${response.code}:\n$body")
+      case Right(body) => println(s"2xx response to GET:\n$body")
+    }
+
+    println("---\n")
+
+    //
+
+    val request2: Request[String, Any] = basicRequest
+      .header("X-Correlation-ID", UUID.randomUUID().toString)
+      .response(asStringAlways)
+      .body("Hello, world!")
+      .post(uri"https://httpbin.org/post")
```

Comment:
Is there a way to avoid writing  the`uri` interpolator and just pass String? E.g. additional method on Request for that. I think it could simplify the usage further for some basic cases.

Reply:
I would prefer not to - people will end up using string interpolation to create URIs, which quickly leads to problems with encoding URI fragments etc. The whole point of the `uri` interpolator is that it is context-aware, and treats anything that is interpolated appropriately as a path/query/etc. fragment.

Reply:
I agree. However, it still leaves me wondering if there is a way to simplify that, as it is the only element that has limited visibility - all the rest can be imported via autosuggestions and has 1:1 name in code - name in import relation.

Reply:
Oh, ok. I had a problem when importing it with Metals, but I agree that most developers are going to use IntelliJ, and if it works great there then that's not an issue.

Reply:
Could we define `uri` as an extension method instead of a method inside an implicit class? Then one would import `sttp.client3.uri` instead of `sttp.client3.UriContext`, which would feel like less magic.

Reply:
You mean in the Scala 3 version instead of https://github.com/softwaremill/sttp-model/blob/master/core/src/main/scala/sttp/model/UriInterpolator.scala#L10 ?

I didn't know interpolators could be added as an extension methods, but I guess it makes sense. 

Anyway, wouldn't that break binary compatibility? (plus the docs are generated using Scala 2 so you'd end up having different code for Scala 2 / Scala 3)

Reply:
Removing the old one and adding a new one would be breaking the binary compatibility, there is a possibility of adding a new one next to the old one at the cost of possible confusion. In Scala 2 sources there would be a need to hack a bit and make a new implicit class called just `uri` that would delegate to the old code.

Reply:
Ah of course, creating an `uri` class would do the trick. I like the idea, but I think we should do it in sttp4, adding it here would have small usability impact and as you write - might be confusing in the import suggestiosn.